pkg/proxy/healthcheck/service_health.go

```diff
@@ -226,14 +229,15 @@ func (h hcHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 	count := svc.endpoints
+	healthzHealthy := h.hcs.healthzServer.IsHealthy()
 	h.hcs.lock.RUnlock()
 
 	resp.Header().Set("Content-Type", "application/json")
 	resp.Header().Set("X-Content-Type-Options", "nosniff")
-	if count == 0 {
-		resp.WriteHeader(http.StatusServiceUnavailable)
-	} else {
+	if count != 0 && healthzHealthy {
```

Comment:
You should probably update the JSON body as well. eg, add a `"kubeProxyHealthy": %v`, to make it clear why it's failing in this case

Reply:
Updated!